actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -56,8 +56,12 @@ def assert_response(type, message = nil)
       #   assert_redirected_to "/some/path", status: :moved_permanently
       def assert_redirected_to(url_options = {}, options = {}, message = nil)
         options, message = {}, options unless options.is_a?(Hash)
-
         status = options[:status] || :redirect
+
+        if AssertionResponse.new(status).code in ...299 | 400... | /\A[1245]/
```

Comment:
Pattern matching is pretty cool, but I think we could simplify this:

```suggestion
        unless AssertionResponse.new(status).code.to_s.match?(/\A3..\z/)
```

Or another possibility would be to put part of this logic in a `AssertionResponse#code_group` method.  Then we could write something like:

```suggestion
        unless AssertionResponse.new(status).code_group == "3XX"
```

Though I'm not sure if there are additional use cases to justify that.